cloudpathlib/s3/s3client.py

```diff
@@ -84,26 +84,26 @@ def _is_file_or_dir(self, cloud_path: S3Path) -> Optional[str]:
         if not cloud_path.key:
             return "dir"
 
-        try:
-            obj = self.s3.ObjectSummary(cloud_path.bucket, cloud_path.key)
-            obj.get()
-            return "file"
-        except self.client.exceptions.NoSuchKey:
-            prefix = cloud_path.key
-            if prefix and not prefix.endswith("/"):
-                prefix += "/"
+        # short circuit files/dirs that are already in the cache
+        if cloud_path._local.exists():
+            return "file" if cloud_path._local.is_file() else "dir"
```

Comment:
What happens if the cloud copy changed and the local cache is out of date?

Reply:
Good point, there is a correctness/speed tradeoff here. I think this only goes wrong if a dir becomes a file or a file becomes a dir. I _think_ that's a pretty unlikely edge case. I'd be willing to skip a network call with that assumption, but could easily be convinced the network call is worth it for correctness. Thoughts?

Reply:
I think another scenario, based on how these functions are called:

- path exists either has a file or dir
- path gets cached locally
- cloud file(s) gets deleted through independent process
- `cp.exists()` will return `False` appropriately, but whichever `cp.is_dir()` and `cp.is_file()` was true before will still return `True` even though file no longer exists

Reply:
Yeah, files getting removed from the server could also cause unexpected behavior. I'll remove this optimization for now since we don't use this for existence anymore and we don't do file/dir checks in a lot of the cache management code so gains of not making the network call are probably pretty small.